apps/obsidian/src/utils/supabaseContext.ts

```diff
@@ -0,0 +1,142 @@
+import {
+  fetchOrCreateSpaceDirect,
+  fetchOrCreatePlatformAccount,
+  createLoggedInClient,
+} from "@repo/database/lib/contextFunctions";
+import type DiscourseGraphPlugin from "~/index";
+
+type Platform = Enums<"Platform">;
+
+export type SupabaseContext = {
+  platform: Platform;
+  spaceId: number;
+  userId: number;
+  spacePassword: string;
+};
+
+let contextCache: SupabaseContext | null = null;
+
+const generateAccountLocalId = (vaultName: string): string => {
+  const randomSuffix = Math.random()
+    .toString(36)
+    .substring(2, 8)
+    .toUpperCase();
+  return `${vaultName}-${randomSuffix}`;
+};
+
+const getOrCreateSpacePassword = async (
+  plugin: DiscourseGraphPlugin,
+): Promise<string> => {
+  if (plugin.settings.spacePassword) {
+    return plugin.settings.spacePassword;
+  }
+  // Generate UUID using crypto.randomUUID()
+  const password = crypto.randomUUID();
+  plugin.settings.spacePassword = password;
+  await plugin.saveSettings();
+  return password;
+};
+
+const getOrCreateAccountLocalId = async (
+  plugin: DiscourseGraphPlugin,
+  vaultName: string,
+): Promise<string> => {
+  if (plugin.settings.accountLocalId) {
+    return plugin.settings.accountLocalId;
+  }
+  const accountLocalId = generateAccountLocalId(vaultName);
+  plugin.settings.accountLocalId = accountLocalId;
+  await plugin.saveSettings();
+  return accountLocalId;
+};
+
+export const getSupabaseContext = async (
+  plugin: DiscourseGraphPlugin,
+): Promise<SupabaseContext | null> => {
+  if (contextCache === null) {
+    try {
+      // Get vault name - try getName() first, fallback to adapter path
+      let vaultName: string;
+      if (typeof plugin.app.vault.getName === "function") {
+        vaultName = plugin.app.vault.getName();
+      } else {
+        // Fallback: use adapter basePath and extract folder name
+        const adapter = plugin.app.vault.adapter;
+        if (adapter && "basePath" in adapter) {
+          const pathParts = (adapter.basePath as string).split(/[/\\]/);
+          vaultName = pathParts[pathParts.length - 1] || "obsidian-vault";
+        } else {
+          vaultName = "obsidian-vault";
+        }
+      }
+      if (!vaultName) {
+        throw new Error("Could not get vault name");
+      }
+
+      const spacePassword = await getOrCreateSpacePassword(plugin);
+      const accountLocalId = await getOrCreateAccountLocalId(plugin, vaultName);
+
+      // Space URL format: "space" + accountLocalId
+      const url = `space${accountLocalId}`;
+      const platform: Platform = "Obsidian";
+
+      const spaceResult = await fetchOrCreateSpaceDirect({
+        password: spacePassword,
+        url,
+        name: vaultName,
+        platform,
+      });
+
+      if (!spaceResult.data) {
+        console.error("Failed to create space");
+        return null;
+      }
+
+      const spaceId = spaceResult.data.id;
+      const userId = await fetchOrCreatePlatformAccount({
```

Comment:
I'm following Roam pattern here to create both Anonymous User and the real user (name here is vaultName instead of "Anonymous of space [x]". But I would say we don't need this for Obsidian since each vault is a space, and is owned by only one person? hence all the handling would go through only need to go through 1 account? I can just comment out this part of the code to delete this real user creation